engine/src/main/java/org/hibernate/validator/internal/engine/ConstraintViolationImpl.java

```diff
@@ -329,9 +329,6 @@ public String toString() {
 	private int createHashCode() {
 		int result = interpolatedMessage != null ? interpolatedMessage.hashCode() : 0;
 		result = 31 * result + ( propertyPath != null ? propertyPath.hashCode() : 0 );
-		result = 31 * result + System.identityHashCode( rootBean );
-		result = 31 * result + System.identityHashCode( leafBeanInstance );
-		result = 31 * result + System.identityHashCode( value );
```

Comment:
Could you try to benchmark with and without it?

From my previous experiments, I often had `System.identityHashCode()` in my profiling but it was more of an artifact.

Reply:
Do you mean just with this change ? or a simple before the changes from this PR ? If it's the later one, than here's a result:
![old_hash_impl](https://user-images.githubusercontent.com/4004823/33011931-af1b03d2-cddf-11e7-9fdd-510a09183a94.png)
I've attached it to the JIRA but not here ...

Reply:
I would like to see the results with all your changes except this one. From what I've seen in my benchmarking campaings (and Sanne confirmed it), `System.identityHashCode()` is often high in the profile for now good reasons. So I'm wondering what would be the result if we keep them here but apply all your other changes.